randovania/game_description/world/area.py

```diff
@@ -23,19 +23,29 @@ def __repr__(self):
         return "Area[{}]".format(self.name)
 
     def __hash__(self):
-        return self.area_asset_id
+        return hash(self.name)
+
+    @property
+    def in_dark_aether(self) -> Optional[bool]:
+        return self.extra.get("in_dark_aether")
+
+    @property
+    def area_asset_id(self) -> int:
+        return self.extra["asset_id"]
```

Comment:
this should either be optional, and use `get()`, or raise a more detailed error if called on an area with no asset ID